Los objetivos de hoy son:

1- Buscar un conjunto de datos a analizar:

Se recomienda que el conjunto de datos a analizar tenga variables numéricas y categóricas. Ya que también se analizarán.

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('../datos/bank-full.csv')
df.sample(3)

age            job  marital  education default  balance housing loan  \
2902    54        retired  married  secondary      no      918     yes   no   
42538   64  self-employed  married   tertiary      no      661      no   no   
7500    38     management   single   tertiary      no     6836     yes   no   

        contact  day month  duration  campaign  pdays  previous poutcome    y  
2902    unknown   14   may       189         1     -1         0  unknown   no  
42538  cellular   21   dec       367         2    294         2  failure  yes  
7500    unknown   29   may       374         4     -1         0  unknown   no

2- Extraer los siguentes estadísiticos del conjunto de datos para varias de las variables disponibles:
- Media
- Desviación con respecto de la media
- Varianza
- Desviación típica
- Moda
- Mediana
- Media ponderada
- Robustez
- Coeficiente de variación

Descripción de los usuarios segun la edad:

In [43]:
media = df['age'].mean()
desv_media = np.abs(df['age'] -media)
var = df['age'].var()
desv_tipica = df['age'].std()
moda = df['age'].mode()[0]
mediana = df['age'].median()
coef_variacion = desv_tipica/media

In [10]:
# creamos pesos aleatorios para poder hacer una media ponderada
w = np.random.randint(0,100, df.shape[0])
media_ponderada = np.average(df['age'], weights=w)

In [ ]:
# robustez = como se calcula ¿?

In [44]:
print(f'media = {media}')
print(f'varianza = {var}')
print(f'desviacion tipica = {desv_tipica}')
print(f'moda = {moda}')
print(f'mediana = {mediana}')
print(f'coeficiente de variacion = {coef_variacion}')
print(f'media ponderada = {media_ponderada}')

media = 40.93621021432837
varianza = 112.75810728286069
desviacion tipica = 10.618762040975431
moda = 32
mediana = 39.0
coeficiente de variacion = 0.25939777974998485
media ponderada = 40.95058260479773


In [9]:
print(f'desv_media = {desv_media}')

desv_media = 0        17.06379
1         3.06379
2         7.93621
3         6.06379
4         7.93621
           ...   
45206    10.06379
45207    30.06379
45208    31.06379
45209    16.06379
45210     3.93621
Name: age, Length: 45211, dtype: float64


Descripción segun la duracion:

Con respecto al empleo:

In [50]:
trabajos = df['job'].value_counts()
df_empleo = pd.DataFrame(trabajos).T

In [51]:
df_empleo

blue-collar  management  technician  admin.  services  retired  \
job         9732        9458        7597    5171      4154     2264   

     self-employed  entrepreneur  unemployed  housemaid  student  unknown  
job           1579          1487        1303       1240      938      288

Datos segun duracion de la llamada:

In [52]:
media2 = df['duration'].mean()
desv_media2 = np.abs(df['duration'] -media)
var2 = df['duration'].var()
desv_tipica2 = df['duration'].std()
moda2 = df['duration'].mode()[0]
mediana2 = df['duration'].median()
coef_variacion2 = desv_tipica2/media2
media_ponderada2 = np.average(df['duration'], weights=w)

In [53]:
print(f'media = {media2}')
print(f'varianza = {var2}')
print(f'desviacion tipica = {desv_tipica2}')
print(f'moda = {moda2}')
print(f'mediana = {mediana2}')
print(f'coeficiente de variacion = {coef_variacion2}')
print(f'media ponderada = {media_ponderada2}')

media = 258.1630797814691
varianza = 66320.57409008514
desviacion tipica = 257.52781226517095
moda = 124
mediana = 180.0
coeficiente de variacion = 0.9975392782080386
media ponderada = 258.80652402847755


In [54]:
print(f'desv_media = {desv_media2}')

desv_media = 0         220.06379
1         110.06379
2          35.06379
3          51.06379
4         157.06379
            ...    
45206     936.06379
45207     415.06379
45208    1086.06379
45209     467.06379
45210     320.06379
Name: duration, Length: 45211, dtype: float64


3- Interpretación de los resultados.
- Ahora interpreta los resultados obtenidos de los diferentes estadísticos estudiados haciendo hincapié en aspectos interesantes de tus datos y explica lo mejor posible la naturaleza de los mismos.

### Con respecto a la edad
La media de la edad es de unos 40/41 años. La mediana está proxima a esta misma edad (39), si bien la moda baja un poco el rango (32), siendo esta la edad que más se repite entre los clientes.

La desviación típica de 10, indica que la variabilidad en la edad se mueve en un un rango entre **30 y 50 años**. El coeficiente de variación es bajo (0.259), indicando que estamos ante unos datos hmogeneos.
